src/ResourceManager/Network/Commands.Network/NetworkWatcher/NewAzureNetworkWatcherProtocolConfiguration.cs

```diff
@@ -0,0 +1,78 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using AutoMapper;
+using Microsoft.Azure.Commands.Network.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Commands.ResourceManager.Common.Tags;
+using Microsoft.Azure.Management.Network;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Network.NetworkWatcher
+{
+    [Cmdlet(VerbsCommon.New, "AzureRmNetworkWatcherProtocolConfiguration"),
+        OutputType(typeof(PSNetworkWatcherProtocolConfiguration))]
+    public class NewAzureNetworkWatcherProtocolConfiguration : NetworkBaseCmdlet
+    {
+        [Parameter(
+             Mandatory = false,
+             ValueFromPipeline = true,
+             HelpMessage = "Procotol")]
+        [ValidateNotNullOrEmpty]
+        [PSArgumentCompleter("Tcp", "Http", "Https", "Icmp")]
+        public string Protocol { get; set; }
+
+        [Parameter(
+             Mandatory = false,
+             ValueFromPipeline = true,
+             HelpMessage = "Method")]
+        [ValidateNotNullOrEmpty]
+        public string Method { get; set; }
+
+        [Parameter(
+             Mandatory = false,
+             ValueFromPipeline = true,
+             HelpMessage = "Header")]
+        [ValidateNotNullOrEmpty]
+        public IDictionary Header { get; set; }
+
+        [Parameter(
+             Mandatory = false,
+             ValueFromPipeline = true,
+             HelpMessage = "ValidStatusCode")]
+        [ValidateNotNullOrEmpty]
+        public int[] ValidStatusCode { get; set; }
+
+        public override void Execute()
+        {
+            base.Execute();
+
+            if (this.Protocol == null)
```

Comment:
@junyezhu this check shouldn't be needed since you have the `ValidateNotNullOrEmpty` attribute on the `Protocol` parameter

Reply:
I think ValidateNotNullOrEmpty will check if user type in -protocol, but no value. I made this check to forbid that user didn't even type in -protocol.

Reply:
@junyezhu if the user doesn't provide a value for a parameter, then it is `null`. You should make the `Protocol` parameter mandatory, and that way you know that they are always providing a non-empty and non-null string due to the `ValidateNotNullOrEmpty` attribute

Reply:
Do you know the keyword to make Protocol parameter mandatory?

Reply:
@junyezhu The `Mandatory` property in the `Parameter` attribute for Protocol property is set to false. Just set it to true. About 30 lines up from here.

Reply:
@MiYanni Fixed. Thanks

Reply:
@junyezhu you can remove this if-statement now